### PR TITLE
[cRO] July, 2021 update

### PR DESCRIPTION
**Pull request by alisonrag**
update map_login
rodex_open_mailbox
rodex_refresh_maillist

### CODE REVIEW

**Comment by hubertyang88** [comment]
bot can log in game , but can't attack by using this cro.pm
usable file attached here
[cRO.zip](https://github.com/OpenKore/openkore/files/6796879/cRO.zip)

**Comment by alisonrag** [comment]
> bot can log in game , but can't attack by using this cro.pm
> usable file attached here
> [cRO.zip](https://github.com/OpenKore/openkore/files/6796879/cRO.zip)

image

latest clean openkore can attack

**Comment by c4c1n6kr3m1** [comment]
@hubertyang88  what do u mean by can not attack? more specific

**Comment by hubertyang88** [comment]
after some more test , this works for latest openkore . 
 I  just update partly of my openkore that might be the cause .

**Comment by hubertyang88** [comment]
I figure it out , I used old config which doesn't have  attackRouteMaxPathDistance 13 , so bot don't attck

**Comment by alisonrag** [comment]
@sctnightcore @ya4ept why this pull is not checked yet?

File: src/Network/Send/cRO.pm [hunk 1]
```diff
@@ -115,4 +123,26 @@ sub sendCharCreate {
...
+	$sex = 0 if ($sex > 1 || $sex < 0); # Sex can only be 0 (female) or 1 (male)
+
+	#my $unknown = pack('C4', (0x34, 0x0B, 0x01, 0x00));
+
+	$msg = $self->reconstruct({
+		switch => 'map_login',
+		accountID => $accountID,
+		charID => $charID,
+		sessionID => $sessionID,
+		unknown => 68404,# 34 0B 01 00
```
Comment (ya4ept) [inline src/Network/Send/cRO.pm:139]: I got `AB 0B 01 00`

File: tables/servers.txt [hunk 2]
```diff
@@ -57,7 +57,7 @@ ignoreAntiCheatWarning 0
 
 # http://ro.zhaouc.com/
 [cRO - China: Taren]
```
Comment (ya4ept) [inline tables/servers.txt:59]: 隆德拉:
`[cRO - China: Lundra]`

File: src/Network/Send/cRO.pm (hunk 1 above)
Comment (alisonrag) [reply src/Network/Send/cRO.pm:139]: maybe has changed, from discord:
image

**Comment by alisonrag** [comment]
https://github.com/OpenKore/openkore/pull/3455/checks?check_run_id=3457614202

**Comment by ya4ept** [comment]
> https://github.com/OpenKore/openkore/pull/3455/checks?check_run_id=3457614202

why is there no error highlighting in windows test?
image

File: src/Network/Send/cRO.pm (hunk 1 above)
Comment (hubertyang88) [reply src/Network/Send/cRO.pm:138]: it seems the server don't check these unknown 4 bytes , bot will log in as long as recvpackets.txt is right.

Comment (ya4ept) [reply src/Network/Send/cRO.pm:138]: @hubertyang88, what values do you have now?

Comment (hubertyang88) [reply src/Network/Send/cRO.pm:138]: I use this one '68404'  , but I tried 0 or 77777 or 99999, all work , I guess server don't care this data , at least for now

Comment (ya4ept) [reply src/Network/Send/cRO.pm:138]: can you intercept packets from the client?

Comment (hubertyang88) [reply src/Network/Send/cRO.pm:138]: 36 04 3d f3 01 00 c4 bc 05 00 06 03 00 00 26 0b 01 00 86 bf 9b 01 00

Comment (ya4ept) [reply src/Network/Send/cRO.pm:138]: i am getting now
36 04 AC 1B 03 00 DC CE 04 00 2D 34 00 00 BF 0B 01 00 D7 23 7F 2A 00
here is a constant: `0B 01 00`

ok, let's leave this part of the code unchanged

File: tables/servers.txt (hunk 2 above)
Comment (alisonrag) [reply tables/servers.txt:59]: @hubertyang88 
can you confirm that this is the server name?
`隆德拉:
[cRO - China: Lundra]`

Comment (hubertyang88) [reply tables/servers.txt:59]: server name is 普隆德拉 = prontera  , the main city name

Comment (ya4ept) [reply tables/servers.txt:59]: sorry i lost the first letter

**Comment by alisonrag** [comment]
seems ok to me

**Comment by ya4ept** [comment]
@hubertyang88, please check the Chinese characters in the files

**Comment by hubertyang88** [comment]
> @hubertyang88, please check the Chinese characters in the files
most of the files look good
In statusnametable.txt line 449 to line 472 , those status are in English , I searched for same status name in deleted lines , those deleted Chinese names are right

**Comment by ya4ept** [comment]
i think now we are ready to merge

**Review by ya4ept — APPROVED** [review]

**Comment by hubertyang88** [comment]
Tested briefly, all looks good.  Marvelous work! Thank you guys!

**Pull request merged**